README.md

```diff
@@ -78,7 +78,19 @@ services:
     environment:
       - TZ=Australia/Perth
       - SBSHOST=readsb
-      - SHARECODE=zg84632abhf231
+      - CLIENTKEY=zg84632abhf231
+```
+
+### Escaping special characters
+
+The ADSBHub client key is full of special characters, that can be misinterpreted on multiple levels. To avoid that:
+* Wrap the environment variable assignment (the whole assignment, not just the key) in single quotation marks so that YAML parses it correctly.
+* Duplicate every `$` character. The single `$` sign is the start of a [variable substitution](https://docs.docker.com/compose/compose-file/#variable-substitution) in docker-compose. Use `$$` instead.
```

Comment:
Thanks for this. Question - do you know if escaping the special characters works? For example `\$`?

Reply:
In YAML:
 - If you use unquoted strings, `\` does not work.
 - It works if double quotes `"` are used.
 - Single quotes `'` are the best because the only character that needs to escaped is another `'`. You'd do that by doubling, not by prefixing it with a `\`. So `abc'ABC` is represented by `'abc''ABC'` in YAML.

The only special character in my client key that made docker compose fail was the `$` but it might not be the only one.
I don't think (have not tried) `\$` works otherwise the [documentation](https://docs.docker.com/compose/compose-file/#variable-substitution) would mention it.

Reply:
Good information, thanks. :-)